mlir/include/mlir/Dialect/Bufferization/IR/BufferizableOpInterface.td

```diff
@@ -575,6 +575,28 @@ def BufferizableOpInterface : OpInterface<"BufferizableOpInterface"> {
           return false;
         }]
       >,
+      InterfaceMethod<
+        /*desc=*/[{
+          Return "true" if the this op has tensor semantics and should be
+          bufferized. By default, ops with tensor operands, tensor op results
+          and/or tensor block arguments have tensor semantics.
+
+          This interface methods can be implemented by ops that should be
+          bufferized but do not have tensor semantics according to the above
+          definition. E.g., this function can return "true" for symbols.
+
+          TODO: This interface method should be called `hasTensorSemantics`, but
```

Comment:
And there are cases where one is true while the other isn't?

Reply:
No, these two functions should never have contradicting return values. (But not all ops that can be bufferized are also destination-style ops, so some ops do not have a `DestinationStyleOpInterface::hasTensorSemantics` function.)

Reply:
Oh, there is actually a difference:
```
    /// Return whether the op has tensor semantics. That is the case if the op
    /// has no memref operands and at least one ranked tensor operand.
```
We may want to drop the part that checks for memrefs operands. (We also have `hasBufferSemantics` and it would make sense to me if both returned "true" when there are mixed tensors/memref; this is somewhat of an edge case, we usually don't have mixed tensors/memrefs.)